### PR TITLE
Feat/broadcast server

### PR DESCRIPTION
**Pull request by chaitanyapotti**
Start using broadcast server for cases where 3pc cookies are not supported

### CODE REVIEW

File: src/utils/StorageHelper.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    this.isInitialized = true;
+  }
+
+  async storeLoginDetails(params: LoginDetails, scope: string): Promise<void> {
+    if (!this.isInitialized) throw new Error("StorageHelper is not initialized");
+    if (this.currentStorageMethod === REDIRECT_PARAMS_STORAGE_METHOD.SERVER) {
+      const privKey = keccak256(scope);
+      const privKeyHex = privKey.toString("hex");
+      const publicKeyHex = getPublic(privKey).toString("hex");
+      const encData = await encryptData(privKeyHex, params);
```
Comment (himanshuchawla009) [inline src/utils/StorageHelper.ts:39]: encryption should be done using public key.

File: src/utils/StorageHelper.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      // use local storage as default for storing stuff
+      this.currentStorageMethod = REDIRECT_PARAMS_STORAGE_METHOD.LOCAL_STORAGE;
+    } else {
+      // use server store as default for storing stuff
+      this.currentStorageMethod = REDIRECT_PARAMS_STORAGE_METHOD.SERVER;
+    }
+    this.isInitialized = true;
+  }
+
+  async storeLoginDetails(params: LoginDetails, scope: string): Promise<void> {
```
Comment (himanshuchawla009) [inline src/utils/StorageHelper.ts:33]: the nonce that we are using as scope here is a randomId is marked to not use for cryptographic purposes. So we should avoid using that.

https://github.com/torusresearch/CustomAuth/blob/b1f7062a8bbcc9961ad0839570346a2e0f1d8d83/src/utils/helpers.ts#L45

**Review by himanshuchawla009 — CHANGES_REQUESTED** [review]

File: src/utils/StorageHelper.ts (hunk 1 above)
Comment (chaitanyapotti) [reply src/utils/StorageHelper.ts:39]: the method internally derives the public key and uses it to encrypt.
it's just a helper function

File: src/utils/StorageHelper.ts (hunk 2 above)
Comment (chaitanyapotti) [reply src/utils/StorageHelper.ts:33]: That's correct. Hence, we hash it and use it since we need a deterministic key before & after redirect

Comment (himanshuchawla009) [reply src/utils/StorageHelper.ts:33]: got it

**Review by himanshuchawla009 — APPROVED** [review]
lgtm, tested in brave with shields on.

**Pull request merged**